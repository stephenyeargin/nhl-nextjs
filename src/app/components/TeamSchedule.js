'use client';

import React from 'react';
import { PropTypes } from 'prop-types';
import Link from 'next/link';
import TeamLogo from '@/app/components/TeamLogo';
import { formatLocalizedDate, formatLocalizedTime, formatBroadcasts } from '@/app/utils/formatters';

const TeamSchedule = ({ team, fullSeasonSchedule, headerStyle }) => {
  return (
    <div className="overflow-x-auto">
      <table className="statsTable">
        <thead>
          <tr>
            <th style={headerStyle} className="text-left">Date</th>
            <th style={headerStyle} className="text-left">Matchup</th>
            <th style={headerStyle} className="" colSpan={2}>Result</th>
            <th style={headerStyle} className="">Broadcasts</th>
          </tr>
        </thead>
        <tbody>
          {fullSeasonSchedule.games.map((game) => (
            <tr key={game.id}>
              <td>
                <time dateTime={game.startTimeUTC} suppressHydrationWarning={true}>
                  {formatLocalizedDate(game.startTimeUTC)}
                </time>
                {' '}
                <time dateTime={game.startTimeUTC} suppressHydrationWarning={true}>
                  {formatLocalizedTime(game.startTimeUTC)}
                </time>
              </td>
              <td>
                <div className="flex gap-2 items-center">
                  {game.gameType === 1 && (
                    <span className="text-xs p-1 border rounded">Preseason</span>
                  )}
                  <TeamLogo team={game.awayTeam.abbrev !== team.abbreviation ? game.awayTeam.abbrev : game.homeTeam.abbrev } className="h-8 w-8" />
                  <Link href={`/game/${game.id}`} className="underline">{game.awayTeam.placeName.default} @ {game.homeTeam.placeName.default}</Link>                   
                </div>
              </td>
              <td className="text-center">
                {(game.gameState === 'OFF' || game.gameState === 'FINAL') && (
                  <>
                    {game.gameOutcome?.lastPeriodType !== 'REG' ? game.gameOutcome?.lastPeriodType : '' }
                    {(team.abbreviation === game.awayTeam.abbrev && game.awayTeam.score > game.homeTeam.score) ? 'W' : (team.abbreviation === game.homeTeam.abbrev && game.homeTeam.score > game.awayTeam.score) ? 'W' : 'L'}
                  </>
                )}
                {['LIVE', 'CRIT'].includes(game.gameState) && (
                  <span className="p-1 text-xs font-bold rounded text-white bg-red-900 uppercase">Live</span>
                )}
              </td>
              <td>
                {['OFF', 'FINAL', 'LIVE', 'CRIT'].includes(game.gameState) && (
                  <Link href={`/game/${game.id}`} className="underline">
                    {game.awayTeam.abbrev} {game.awayTeam.score}-{game.homeTeam.score} {game.homeTeam.abbrev}
                  </Link>
                )}
                {game.gameScheduleState	 === 'CNCL' && (
                  <span className="p-1 text-xs border rounded">Canceled</span>
                )}
              </td>
              <td>
                {formatBroadcasts(game.tvBroadcasts)}
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

TeamSchedule.propTypes = {
  team: PropTypes.object.isRequired,
  fullSeasonSchedule: PropTypes.object.isRequired,
  headerStyle: PropTypes.object.isRequired,
};

export default TeamSchedule;