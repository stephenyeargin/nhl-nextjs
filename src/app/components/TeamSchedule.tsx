'use client';

import React from 'react';
import Link from 'next/link';
import TeamLogo from '@/app/components/TeamLogo';
import { formatLocalizedDate, formatLocalizedTime, formatBroadcasts } from '@/app/utils/formatters';
import styles from '@/app/components/StatsTable.module.scss';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faBan, faWarning } from '@fortawesome/free-solid-svg-icons';

interface TeamInfo {
  abbreviation: string;
}

interface PlaceName {
  default: string;
}

interface GameTeam {
  abbrev: string;
  score: number;
  placeName: PlaceName;
}

interface GameOutcome {
  lastPeriodType?: string;
}

interface BroadcastInfo { network: string; market: string }

interface Game {
  id: number | string;
  startTimeUTC: string;
  gameType: number;
  gameState: string;
  gameScheduleState?: string;
  gameOutcome?: GameOutcome;
  awayTeam: GameTeam;
  homeTeam: GameTeam;
  tvBroadcasts?: BroadcastInfo[] | null;
}

interface FullSeasonSchedule {
  games: Game[];
}

interface TeamScheduleProps {
  team: TeamInfo;
  fullSeasonSchedule: FullSeasonSchedule;
  headerStyle?: React.CSSProperties;
}

const TeamSchedule = ({ team, fullSeasonSchedule, headerStyle }: TeamScheduleProps) => {
  return (
    <div className="overflow-x-auto">
  <table className={styles.statsTable}>
        <thead>
          <tr>
            <th style={headerStyle} className="text-left">Date</th>
            <th style={headerStyle} className="text-left">Matchup</th>
            <th style={headerStyle} className="" colSpan={2}>Result</th>
            <th style={headerStyle} className="">Broadcasts</th>
          </tr>
        </thead>
        <tbody>
          {fullSeasonSchedule.games.map((game) => (
            <tr key={game.id}>
              <td>
                <time dateTime={game.startTimeUTC} suppressHydrationWarning={true}>
                  {formatLocalizedDate(game.startTimeUTC)}
                </time>
                {' '}
                <time dateTime={game.startTimeUTC} suppressHydrationWarning={true}>
                  {formatLocalizedTime(game.startTimeUTC)}
                </time>
              </td>
              <td>
                <div className="flex gap-2 items-center">
                  {game.gameType === 1 && (
                    <span className="text-xs p-1 border rounded">Preseason</span>
                  )}
                  <TeamLogo team={game.awayTeam.abbrev !== team.abbreviation ? game.awayTeam.abbrev : game.homeTeam.abbrev } className="h-8 w-8" />
                  <Link href={`/game/${game.id}`} className="underline">{game.awayTeam.placeName.default} @ {game.homeTeam.placeName.default}</Link>                   
                </div>
              </td>
              <td className="text-center">
                {(game.gameState === 'OFF' || game.gameState === 'FINAL') && (
                  <>
                    {game.gameOutcome?.lastPeriodType !== 'REG' ? game.gameOutcome?.lastPeriodType : '' }
                    {(team.abbreviation === game.awayTeam.abbrev && game.awayTeam.score > game.homeTeam.score) ? 'W' : (team.abbreviation === game.homeTeam.abbrev && game.homeTeam.score > game.awayTeam.score) ? 'W' : 'L'}
                  </>
                )}
                {['LIVE', 'CRIT'].includes(game.gameState) && (
                  <span className="p-1 text-xs font-bold rounded text-white bg-red-900 uppercase">Live</span>
                )}
              </td>
              <td>
                {['OFF', 'FINAL', 'LIVE', 'CRIT'].includes(game.gameState) && (
                  <Link href={`/game/${game.id}`} className="underline">
                    {game.awayTeam.abbrev} {game.awayTeam.score}-{game.homeTeam.score} {game.homeTeam.abbrev}
                  </Link>
                )}
                {game.gameScheduleState === 'CNCL' && (
                  <span className="text-sm font-medium px-2 py-1 bg-yellow-500 text-black rounded mr-1 uppercase"><FontAwesomeIcon icon={faBan} fixedWidth /> Cancelled</span>
                )}
                {game.gameScheduleState === 'PPD' && (
                  <span className="text-sm font-medium px-2 py-1 bg-yellow-500 text-black rounded mr-1 uppercase"><FontAwesomeIcon icon={faWarning} fixedWidth /> Postponed</span>
                )} 
              </td>
              <td>
                {formatBroadcasts(game.tvBroadcasts)}
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default TeamSchedule;