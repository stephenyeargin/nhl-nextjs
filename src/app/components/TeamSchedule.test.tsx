import React from 'react';
import { render, screen } from '@testing-library/react';
import TeamSchedule from './TeamSchedule';

jest.mock('next/link', () => {
  const MockLink = ({ children, href }: any) => <a href={href}>{children}</a>;
  MockLink.displayName = 'MockLink';

  return MockLink;
});
jest.mock('@/app/components/TeamLogo', () => {
  const MockTeamLogo = ({ team }: any) => <div data-testid={`logo-${team}`} />;
  MockTeamLogo.displayName = 'MockTeamLogo';

  return MockTeamLogo;
});
jest.mock('@/app/utils/formatters', () => ({
  formatLocalizedDate: () => 'Jan 1',
  formatLocalizedTime: () => '7:00 PM',
  formatBroadcasts: (b: any) => (b && b.length ? 'NET (NAT)' : 'No Broadcasts'),
}));

const gameBase = {
  id: 1,
  startTimeUTC: '2025-01-01T00:00:00Z',
  gameType: 2,
  gameState: 'FINAL',
  awayTeam: { abbrev: 'AWY', score: 2, placeName: { default: 'Away' } },
  homeTeam: { abbrev: 'HOM', score: 3, placeName: { default: 'Home' } },
};

describe('TeamSchedule', () => {
  test('renders final game with W/L logic and broadcasts', () => {
    const team = { abbreviation: 'HOM' };
    render(
      <TeamSchedule
        team={team}
        fullSeasonSchedule={{
          games: [
            {
              ...gameBase,
              tvBroadcasts: [{ network: 'NET', market: 'NAT' }],
              gameOutcome: { lastPeriodType: 'OT' },
            },
          ],
        }}
      />
    );
    expect(screen.getByText(/OT/)).toBeTruthy();
    expect(screen.getByText('NET (NAT)')).toBeTruthy();
  });

  test('renders live game badge and cancelled', () => {
    const team = { abbreviation: 'AWY' };
    render(
      <TeamSchedule
        team={team}
        fullSeasonSchedule={{
          games: [
            { ...gameBase, id: 2, gameState: 'LIVE' },
            { ...gameBase, id: 3, gameState: 'FUT', gameScheduleState: 'CNCL' },
            { ...gameBase, id: 4, gameState: 'FUT', gameScheduleState: 'PPD' },
          ],
        }}
      />
    );
    expect(screen.getByText(/Live/i)).toBeTruthy();
    expect(screen.getByText(/Cancelled/)).toBeTruthy();
    expect(screen.getByText(/Postponed/)).toBeTruthy();
  });
});
