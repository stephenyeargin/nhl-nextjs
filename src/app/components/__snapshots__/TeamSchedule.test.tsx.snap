// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamSchedule renders final game with W/L logic and broadcasts 1`] = `
<DocumentFragment>
  <div
    class="overflow-x-auto"
  >
    <table
      class="statsTable"
    >
      <thead>
        <tr>
          <th
            class="text-left"
          >
            Date
          </th>
          <th
            class="text-left"
          >
            Matchup
          </th>
          <th
            class=""
            colspan="2"
          >
            Result
          </th>
          <th
            class=""
          >
            Broadcasts
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <time
              datetime="2025-01-01T00:00:00Z"
            >
              Jan 1
            </time>
             
            <time
              datetime="2025-01-01T00:00:00Z"
            >
              7:00 PM
            </time>
          </td>
          <td>
            <div
              class="flex gap-2 items-center"
            >
              <div
                data-testid="logo-AWY"
              />
              <a
                href="/game/1"
              >
                Away @ Home
              </a>
            </div>
          </td>
          <td
            class="text-center"
          >
            OTW
          </td>
          <td>
            <a
              href="/game/1"
            >
              AWY 2-3 HOM
            </a>
          </td>
          <td>
            NET (NAT)
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`TeamSchedule renders live game badge and cancelled 1`] = `
<DocumentFragment>
  <div
    class="overflow-x-auto"
  >
    <table
      class="statsTable"
    >
      <thead>
        <tr>
          <th
            class="text-left"
          >
            Date
          </th>
          <th
            class="text-left"
          >
            Matchup
          </th>
          <th
            class=""
            colspan="2"
          >
            Result
          </th>
          <th
            class=""
          >
            Broadcasts
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <time
              datetime="2025-01-01T00:00:00Z"
            >
              Jan 1
            </time>
             
            <time
              datetime="2025-01-01T00:00:00Z"
            >
              7:00 PM
            </time>
          </td>
          <td>
            <div
              class="flex gap-2 items-center"
            >
              <div
                data-testid="logo-HOM"
              />
              <a
                href="/game/2"
              >
                Away @ Home
              </a>
            </div>
          </td>
          <td
            class="text-center"
          >
            <span
              class="p-1 text-xs font-bold rounded text-white bg-red-900 uppercase"
            >
              Live
            </span>
          </td>
          <td>
            <a
              href="/game/2"
            >
              AWY 2-3 HOM
            </a>
          </td>
          <td>
            No Broadcasts
          </td>
        </tr>
        <tr>
          <td>
            <time
              datetime="2025-01-01T00:00:00Z"
            >
              Jan 1
            </time>
             
            <time
              datetime="2025-01-01T00:00:00Z"
            >
              7:00 PM
            </time>
          </td>
          <td>
            <div
              class="flex gap-2 items-center"
            >
              <div
                data-testid="logo-HOM"
              />
              <a
                href="/game/3"
              >
                Away @ Home
              </a>
            </div>
          </td>
          <td
            class="text-center"
          />
          <td>
            <span
              class="text-sm font-medium px-2 py-1 bg-yellow-500 text-black rounded mr-1 uppercase"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-ban fa-fw "
                data-icon="ban"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M367.2 412.5L99.5 144.8C77.1 176.1 64 214.5 64 256c0 106 86 192 192 192c41.5 0 79.9-13.1 111.2-35.5zm45.3-45.3C434.9 335.9 448 297.5 448 256c0-106-86-192-192-192c-41.5 0-79.9 13.1-111.2 35.5L412.5 367.2zM0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256z"
                  fill="currentColor"
                />
              </svg>
               Cancelled
            </span>
          </td>
          <td>
            No Broadcasts
          </td>
        </tr>
        <tr>
          <td>
            <time
              datetime="2025-01-01T00:00:00Z"
            >
              Jan 1
            </time>
             
            <time
              datetime="2025-01-01T00:00:00Z"
            >
              7:00 PM
            </time>
          </td>
          <td>
            <div
              class="flex gap-2 items-center"
            >
              <div
                data-testid="logo-HOM"
              />
              <a
                href="/game/4"
              >
                Away @ Home
              </a>
            </div>
          </td>
          <td
            class="text-center"
          />
          <td>
            <span
              class="text-sm font-medium px-2 py-1 bg-yellow-500 text-black rounded mr-1 uppercase"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-triangle-exclamation fa-fw "
                data-icon="triangle-exclamation"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 32c14.2 0 27.3 7.5 34.5 19.8l216 368c7.3 12.4 7.3 27.7 .2 40.1S486.3 480 472 480L40 480c-14.3 0-27.6-7.7-34.7-20.1s-7-27.8 .2-40.1l216-368C228.7 39.5 241.8 32 256 32zm0 128c-13.3 0-24 10.7-24 24l0 112c0 13.3 10.7 24 24 24s24-10.7 24-24l0-112c0-13.3-10.7-24-24-24zm32 224a32 32 0 1 0 -64 0 32 32 0 1 0 64 0z"
                  fill="currentColor"
                />
              </svg>
               Postponed
            </span>
          </td>
          <td>
            No Broadcasts
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
